lib/process-services/src/lib/task-list/services/task-filter.service.ts

```diff
@@ -67,30 +67,32 @@ export class TaskFilterService {
                 (res) => {
                     const filters: FilterRepresentationModel[] = [];
                     res.forEach((filter) => {
-                        if (filter.name === involvedTasksFilter.name) {
-                            filters.push(new FilterRepresentationModel({
-                                ...filter,
-                                filter: involvedTasksFilter.filter,
-                                appId
-                            }));
-                        } else if (filter.name === myTasksFilter.name) {
-                            filters.push(new FilterRepresentationModel({
-                                ...filter,
-                                filter: myTasksFilter.filter,
-                                appId
-                            }));
-                        } else if (filter.name === queuedTasksFilter.name) {
-                            filters.push(new FilterRepresentationModel({
-                                ...filter,
-                                filter: queuedTasksFilter.filter,
-                                appId
-                            }));
-                        } else if (filter.name === completedTasksFilter.name) {
-                            filters.push(new FilterRepresentationModel({
-                                ...filter,
-                                filter: completedTasksFilter.filter,
-                                appId
-                            }));
+                        if (!this.isFilterAlreadyExisting(filters, filter.name)) {
```

Comment:
This new condition should be tested as part of `createDefaultFilters` method as well, to ensure this bug won't happen again

Reply:
This condition is already being checked as a part of createDefaultFilters() methods.

createDefaultFilters() for task filters => https://github.com/Alfresco/alfresco-ng2-components/blob/ACS-5480/lib/process-services/src/lib/task-list/services/task-filter.service.ts#L70
createDefaultFilters() for process-filters => https://github.com/Alfresco/alfresco-ng2-components/blob/ACS-5480/lib/process-services/src/lib/process-list/services/process-filter.service.ts#L121

Reply:
Yes I can see it but you didn't neither add new test case nor modify existing ones that will check what happens if someone will try to create duplicated process filters, this should be covered by unit tests too

Reply:
Hi, I have added unit tests for createDefaultFilter() methods. The tests check if the response of createDefaultFilter() is an array with unique values even in the presence of a duplicate filter.